### ISSUE TITLE
anyone can explain me how to find eta of batch gradient decent using GridSearchCv

### ISSUE THREAD

**Issue by deep3125** [ageron/handson-ml2#139]

**Comment by ageron** [comment]
Hi @deep3125 ,

Here's one approach:

```python
from sklearn.linear_model import SGDClassifier
from sklearn.datasets import make_moons
from sklearn.model_selection import GridSearchCV

param_grid = {"eta0": [0.001, 0.003, 0.01, 0.03, 0.1, 0.3, 1.0]}
sgd_clf = SGDClassifier(learning_rate="constant", eta0=0.001)
grid_search_cv = GridSearchCV(sgd_clf, param_grid)

X_train, y_train = make_moons()
grid_search_cv.fit(X_train, y_train)

print("Best learning rate:", grid_search_cv.best_estimator_.eta0)
```

I recommend you also check out the 1-cycle strategy, described in part 2.

Hope this helps.

**Comment by deep3125** [comment]
Thanks
